plugins/main/host-device/host-device.go

```diff
@@ -278,6 +284,16 @@ func getLink(devname, hwaddr, kernelpath string) (netlink.Link, error) {
 				}
 			}
 		}
+	} else if len(pciaddr) > 0 {
+		netDir := filepath.Join(sysBusPci, pciaddr, "net")
+		if _, err := os.Lstat(netDir); err != nil {
+			return nil, fmt.Errorf("no net directory under pci device %s: %q", pciaddr, err)
+		}
+		fInfo, err := ioutil.ReadDir(netDir)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read net directory %s: %q", netDir, err)
+		}
+		return netlink.LinkByName(fInfo[0].Name())
```

Comment:
There is no check if this list has at least single entry. I'm not sure if this directory can be empty, but if it's possible - code will panic there. If you are sure that it has to have at least a single entry - just leave a comment about that there in code.

Reply:
added check for ``fInfo`` and return netlink by name when ``fInfo`` list larger than 0.